utilities/poap/README-POAP-agent-install.md

```diff
@@ -0,0 +1,132 @@
+# POAP: Puppet Agent Installation & Setup
+#### Table of Contents
+
+1. [Overview](#overview)
+1. [Pre-Install Tasks](#pre-install)
+1. [Script Parameters](#parms)
+1. [Script Execution](#execute)
+1. [Non-POAP Installation Methods](#nonpoap)
+1. [References](#references)
+
+## <a name="overview">Overview</a>
+
+This document describes automated Puppet agent installation and setup on Cisco Nexus switches using PowerOn Auto Provisioning (POAP). See the [Non-POAP Installation](#nonpoap) section for other puppet agent installation methods.
+
+Automated installation is achieved with a python script that performs the following tasks in the agent environment:
+
+1. Set up basic networking and verify reachability
+1. Install the Puppet Agent RPM
+1. Configure `puppet.conf`
+1. Execute an initial puppet agent run to generate SSL certificates
+
+## <a name="pre-install">Pre-Install Tasks</a>
+
+**Platform Requirements**
+
+Please reference the **Pre-Install Tasks** section of the [README-agent-install.md](README-agent-install.md) document for a list of minimum platform and software requirements.
+
+**POAP Infrastructure**
+
+This document focuses on Puppet Agent installation and setup only. For more information regarding POAP Installation and Usage please see [Using PowerOn Auto Provisioning][poap doc]. This document assumes that the appropriate network infrastructure is present for POAP automation, ie. a DHCP server for bootstrapping the cisco device and a script server (TFTP/HTTP) containing device images and configuration scripts.
+
+**Puppet Install Script**
+
+The script is a python file found under the utilities directory:
+
+```
+cisco-network-puppet-module/utilities/poap/puppet_agent_install.py
+```
+
+Copy the script to your script server and modify your POAP configuration script to call `puppet_agent_install.py`. See [Script Parameters](#script) for guidance on modifying the `puppet_agent_install.py` settings to reflect your requirements.
+
+## <a name="parms">Script Parameters</a>
+
+The script has a number of required and optional parameters. Change these parameters in the script or set environment variables to override the script values.
+
+##### MD5 Checksum
+
+Please note that the md5 checksum should be recalculated if any changes are made to the `puppet_agent_install.py` script. The checksum is specified on line 2 of the script; it will be automatically updated by issuing the following command after making any changes:
+
+```
+f=puppet_agent_install.py ; cat $f | sed '/^#md5sum/d' > $f.md5 ; sed -i "s/^#md5sum=.*/#md5sum=$(md5sum $f.md5 | sed 's/ .*//')/" $f
+```
+
+##### Required Parameters
+
+* `RPM_NAME` This is the puppet agent release RPM. Specify the RPM required by your environment:
```

Comment:
I think that hardcoding the rpm names for the example is fine, but let's also include a note indicating users should check `http://yum.puppetlabs.com` for the latest release rpm.